File: src/my_project/easy_problems/from151To200/increasingDecreasingString.py
from typing import List 

class Solution:
    def sortString(self, s: str) -> str:
        vocabulary =  "abcdefghijklmnopqrstuvwxyz"
        dict_vocabulary = dict()
        for letter in vocabulary:
            dict_vocabulary[letter] = 0

        for i in s:
            dict_vocabulary[i] += 1

        answer = ""

        while len(answer) < len(s):

            for i in range(len(vocabulary)):
                if dict_vocabulary[vocabulary[i]] > 0:
                    answer += vocabulary[i]
                    dict_vocabulary[vocabulary[i]] -= 1

            for i in range(len(vocabulary)-1, -1, -1):
                if dict_vocabulary[vocabulary[i]] > 0:
                    answer += vocabulary[i]
                    dict_vocabulary[vocabulary[i]] -= 1

        return answer


solution = Solution()

print(solution.sortString(s = "aaaabbbbcccc")) 

print(solution.sortString(s = "rat")) 

print(solution.sortString(s = "leetcode")) 

print(solution.sortString(s = "ggggggg")) 

File: my_project_typescript/easy_problems_typescript/from51To100/increasingDecreasingString.ts
function sortString(s: string): string {
    let vocabulary: string = "abcdefghijklmnopqrstuvwxyz"
    let dict_vocabulary: any = {}

    for (let letter of vocabulary){
        dict_vocabulary[letter] = 0
    }

    for (let i of s){
        dict_vocabulary[i] += 1
    }

    let answer: string = ""

    while (answer.length < s.length){

        for (let i = 0; i < vocabulary.length; i++){
            if (dict_vocabulary[vocabulary[i]] > 0){
                answer += vocabulary[i]
                dict_vocabulary[vocabulary[i]] -= 1
            }
        }

        for (let i = vocabulary.length-1; i > -1; i = i -1){
            if (dict_vocabulary[vocabulary[i]] > 0){
                answer += vocabulary[i]
                dict_vocabulary[vocabulary[i]] -= 1
            }
        }
    }

    return answer 
};

console.log(sortString( "aaaabbbbcccc"))

console.log(sortString( "rat"))

console.log(sortString( "leetcode"))

console.log(sortString("ggggggg"))
